include/revng/Support/Assert.h

```diff
@@ -199,3 +210,15 @@ void setAbortHook(AbortHook Hook);
 #ifdef __cplusplus
 }
 #endif
+
+#if REVNG_ASSERT_IS_MACRO == 0 && defined(__cplusplus)
+template<typename Arg>
+inline void revng_assert(Arg &&Cond) {
+  revng_assert_hidden(Cond);
+}
+
+template<typename Arg, typename Message>
+inline void revng_assert(Arg &&Cond, Message &&Msg) {
```

Comment:
Inner functions assume `const char *` instead of `Message`.

Reply:
Uhm, you sure? I can understand the other one because we explicitly cast to `bool` in the macro, but here?

Reply:
i think i will delay this commit into a unrelated rewriter mr so this one can go in